src/test/java/software/amazon/documentdb/jdbc/DocumentDbConnectionTest.java

```diff
@@ -490,89 +479,37 @@ void testMultiProcessConnections(final DocumentDbTestEnvironment environment) th
+            final Runner runner = new Runner(internalSSHTunnelProperties);
+            final Thread thread = new Thread(runner);
+            runners.add(runner);
+            threads.add(thread);
         }
-        return processes;
-    }
-
-    private static void assertProcessesCompleteNormally(final List<Process> processes, final Instant timeoutTime)
-            throws InterruptedException, IOException {
-        boolean timeoutReached = false;
-        for (Process process : processes) {
-            synchronized (process) {
-                while (process.isAlive()) {
-                    final Instant now = Instant.now();
-                    if (now.isAfter(timeoutTime)) {
-                        LOGGER.debug("Timeout reached.");
-                        timeoutReached = true;
-                        break;
-                    }
-                    process.waitFor(500, TimeUnit.MILLISECONDS);
-                }
-                if (!timeoutReached) {
-                    LOGGER.debug("Closed before timeout.");
-                }
-                // Forceful shutdown, if necessary.
-                if (process.isAlive()) {
-                    LOGGER.debug("Destroying process.");
-                    process.destroy();
-                }
-                if (process.isAlive()) {
-                    LOGGER.debug("Forcibly destroying process.");
-                    process.destroyForcibly();
-                }
-                try (BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(process.getInputStream(), StandardCharsets.UTF_8))) {
-                    final String stdOut = bufferedReader.lines().collect(Collectors.joining(System.lineSeparator()));
-                    if (!isNullOrWhitespace(stdOut)) {
-                        LOGGER.debug("Process output: '" + stdOut + "'");
-                    }
-                } catch (IOException | UncheckedIOException e) {
-                    // Ignore exceptions - might already be closed.
-                }
-                try (BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(process.getErrorStream(), StandardCharsets.UTF_8))) {
-                    final String stdErr = bufferedReader.lines().collect(Collectors.joining(System.lineSeparator()));
-                    if (!isNullOrWhitespace(stdErr)) {
-                        LOGGER.debug("Process error: '" + stdErr + "'");
-                    }
-                } catch (IOException | UncheckedIOException e) {
-                    // Ignore exceptions - might already be closed.
+        for (final Thread thread : threads) {
+            thread.start();
+        }
+        while (threads.size() > 0) {
+            TimeUnit.MILLISECONDS.sleep(100);
+            for (int i = threads.size() - 1; i >= 0; i--) {
+                final Thread thread = threads.get(i);
+                if (!thread.isAlive()) {
+                    thread.join();
+                    threads.remove(i);
```

Comment:
Should threads do remove here as it could change threads size?

Reply:
This is by design. It will make it easier to know when we no longer have any active threads. I start from the "end" and work to the "beginning" in the loop.